mysql-test/mysql-test-run.pl

```diff
@@ -2838,16 +2839,26 @@ ()
     else
     {
       # hm, what paths work for debs and for rpms ?
-      for (<$bindir/lib64/mysql/plugin/*.so>,
+      #
+      # Note: libmariadb3 is first, as in Debian, we
+      #       separate server plugins from client plugins.
+      for (<$bindir/lib/*/libmariadb3/plugin/*.so>,
+           <$bindir/lib64/mysql/plugin/*.so>,
```

Comment:
No, it's not a good idea. This is the loop for server plugins. It sets environment variables for `INSTALL PLUGIN` and sets `$plugindir`. Neither is needed for client plugins. Better have a separate loop specifically for client plugins.

Reply:
When its moved to a separate loop (sounds familiar), how would you determine `${DEB_HOST_MULTIARCH}` in perl? When Debian come up with multiarch paths I doubt the intent was they should be globed together and as such the `*` in the path should correspond to the host architecture.

Reply:
Ping @cvicentiu, this comment is about your commit.

Reply:
@vuvova: Ok, I did write the separate loop, but I didn't like the code duplication that ensued. Coming up with a cleaner one now.

@grooverdan yes, determining the right DEB_HOST_MULTIARCH is impossible in this current state as you'd get the first one that matches. Additionally, if one decides to really change file locations by changing the cmake config parameters, this thing backfires and mtr simply won't work.

My idea initially was to get rid of this mess entirely and just set the correct directory in a cmake configure step, but I wasn't sure if I can get it to work properly on time, so I "hacked" this libmariadb3 in to get the ball rolling.

Reply:
quite right the entire path determination is because mtr is independent of most build args. Problem for another day.